napari/utils/events/dataclass.py

```diff
@@ -180,13 +182,39 @@ def set_with_events(self: C, name: str, value: Any) -> None:
 
     # if different we emit the event with new value
     after = getattr(self, name)
-    if isinstance(before, np.ndarray) or isinstance(after, np.ndarray):
-        different = np.any(before != after)
-    else:
-        different = before != after
-    if different:
+    if not compare(after, before):
         # use gettattr again in case `_on_name_set` has modified it
         getattr(self.events, name)(value=after)  # type: ignore
+        self.events.values_updated()
+
+
+def compare(v1, v2):
```

Comment:
This function should probably live in its own utils file, maybe `events/utils.py`.

I don't love that this now depends on dask and numpy, but maybe it's unavoidable for now ...

Reply:
> This function should probably live in its own utils file, maybe `events/utils.py`.

I'm not an enthusiast of this. Its do only shallow comparison. Maybe It should be changed to a private method. 

> I don't love that this now depends on dask and numpy, but maybe it's unavoidable for now ...

Its already (earlier) depends on NumPy. I only add dask to avoid expensive calculations.

Reply:
Assuming this comparison functionality stays in this PR, can we rename this function to something that indicates its return value a little more?   something like "`is_equal`" instead of `compare` ... 
also, add a `Returns` section to the numpydocs and move the description of what is returned (and add a type) there:
```
Returns
-------
result : bool
    Returns ``True`` if ``v1`` and ``v2`` are equivalent.  Dask arrays, and any two objects of different ``type`` will always return ``False``..
```

Reply:
Agreed, this would be nicer. With this change I'm then happy to leave in this PR, as it doesn't change current behavior and will fix all our warnings. It can be improved/ changed later.